apps/tutorial-servo-can-control/src/main.cpp

```diff
@@ -0,0 +1,24 @@
+#include "mbed.h"
+#include "TutorialServo.h"
+
+PinName servoPin(PA_1);
+CAN     can(PB_8, PB_9);  // assume rx_pin:PB_8, tx_pin: PB_9
+
+int main(){
+    // initializtion
+    TutorialServo servo_out(servoPin);
+    float rec_can;
+    float output_degree;
+
+    can.frequency(1000000); // set CAN bit rate to 1Mbps
+    CANMessage rxMsg;
+
+    // loop
+    while(1){
+        if (can.read(rxMsg)) {
+            rec_can = rxMsg.data[0];
```

Comment:
CANMessage.data is an array of 8 bytes, so data[0] will only give the first byte of the CAN data payload. This would be fine if your data was only 1 byte long, but it wouldn't work for floats which are 4 bytes long. Check out our custom wrapper class CANMsg (https://github.com/uwrobotics/MarsRoverFirmware/blob/master/libs/can/include/CANMsg.h), I recommend using its getPayload() function instead, eg.
```
CANMsg msg;
float rec_can;
...
// poll for new CAN message
...
msg.getPayload(rec_can);
```